Zend/zend_operators.c

```diff
@@ -54,7 +54,38 @@ static _locale_t current_locale = NULL;
 
 #define TYPE_PAIR(t1,t2) (((t1) << 4) | (t2))
 
-static const unsigned char tolower_map[256] = {
+#if __SSE2__
+#define HAVE_BLOCKCONV
+
+/* Common code for SSE2 accelerated character case conversion */
+
+#define BLOCKCONV_INIT_RANGE(start, end) \
+	const __m128i blconv_start_minus_1 = _mm_set1_epi8((start) - 1); \
+	const __m128i blconv_end_plus_1 = _mm_set1_epi8((end) + 1); \
+	__m128i blconv_operand, blconv_gt, blconv_lt, blconv_mingle;
```

Comment:
Why aren't these declarations in BLOCKCONV_LOAD?

Reply:
It avoids duplicate declarations if you want to have multiple load/store invocations in a single function. Also it gets us closer to the C89 style of putting declarations at the start of a scope block -- I received feedback about that when I submitted a PHP patch a few years ago.

Reply:
PHP no longer uses C89. If you are concerned about redeclarations, you can wrap them in a block scope (the load result would have to be a macro argument in that case).

Reply:
Done.